src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.h

```diff
@@ -0,0 +1,195 @@
+{
+public:
+    /**
+     * @brief Creates a Microwave Oven Control cluster instance. The Init() function needs to be called for this instance
+     * to be registered and called by the interaction model at the appropriate times.
+     * @param aDelegate A pointer to the delegate to be used by this server.
+     * Note: the caller must ensure that the delegate lives throughout the instance's lifetime.
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the Microwave Oven Control cluster to be instantiated.
+     * @param aOpStateInstance The pointer of Operational State Instance.
+     * @param aMicrowaveOvenModeInstance The pointer of Microwave Oven Mode Instance.
+     * Note: a MicrowaveOvenControl instance must live relying on an Operational State instance and a Microwave Oven Mode instance,
+     * caller should be initialized that 2 instances before initializing MicorwaveOvenControl instance.
+     */
+    Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId,
+             Clusters::OperationalState::Instance * aOpStateInstance, Clusters::ModeBase::Instance * aMicrowaveOvenModeInstance);
+
+    ~Instance() override;
+
+    /**
+     * @brief Initialise the Microwave Oven Control server instance.
+     * This function must be called after defining an Instance class object.
+     * @return Returns an error if the given endpoint and cluster ID have not been enabled in zap or if the
+     * CommandHandler or AttributeHandler registration fails, else returns CHIP_NO_ERROR.
+     */
+    CHIP_ERROR Init();
+
+    /**
+     * @brief define the get/set api for the mandatory attributes
+     */
+    uint32_t GetCookTime() const;
+    void SetCookTime(uint32_t cookTime);
+
+    uint8_t GetPowerSetting() const;
+    void SetPowerSetting(uint8_t powerSetting);
+
+private:
+    Delegate * mDelegate;
+    EndpointId mEndpointId;
+    ClusterId mClusterId;
+    /**
+     * Operational State instance
+     */
+    Clusters::OperationalState::Instance * mOpStateInstance = nullptr;
+    /**
+     * Microwave Oven Mode instance
+     */
+    Clusters::ModeBase::Instance * mMicrowaveOvenModeInstance = nullptr;
+
+    /**
+     * set default values
+     */
+    uint32_t mCookTime    = kDefaultCookTime;
+    uint8_t mPowerSetting = kDefaultPowerSetting;
+
+    /**
+     * IM-level implementation of read
+     * @return appropriately mapped CHIP_ERROR if applicable (may return CHIP_IM_GLOBAL_STATUS errors)
+     */
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    /**
+     * @brief Inherited from CommandHandlerInterface
+     */
+    void InvokeCommand(HandlerContext & ctx) override;
+
+    /**
+     * @brief Handle Command: SetCookingParameters.
+     * This programme will also check if the input values is invalid.
```

Comment:
```suggestion
     * This method will also check if the input values is invalid.
```

What happens if it's invalid? There is no return value...